geode-core/src/distributedTest/java/org/apache/geode/cache30/ReconnectWithClusterConfigurationDUnitTest.java

```diff
@@ -0,0 +1,191 @@
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.rules.DistributedRule;
+
+public class ReconnectWithClusterConfigurationDUnitTest implements Serializable {
+  static final int NUM_LOCATORS = 2;
+  static final int NUM_VMS = 4;
+  static DistributedSystem system;
+  static Cache cache;
+  static Locator locator;
+  static int[] locatorPorts = new int[NUM_LOCATORS];
+  static Properties dsProperties;
+
+  @Rule
+  public DistributedRule distributedRule = DistributedRule.builder().withVMCount(NUM_VMS).build();
+
+  @Before
+  public void setup() {
+    List<AvailablePort.Keeper> randomAvailableTCPPortKeepers =
+        AvailablePortHelper.getRandomAvailableTCPPortKeepers(NUM_LOCATORS);
+    for (int i = 0; i < NUM_LOCATORS; i++) {
+      AvailablePort.Keeper keeper = randomAvailableTCPPortKeepers.get(i);
+      locatorPorts[i] = keeper.getPort();
+    }
+    final int[] locPorts = locatorPorts;
+    Invoke.invokeInEveryVM("set locator ports", () -> locatorPorts = locPorts);
+    for (int i = 0; i < NUM_LOCATORS; i++) {
+      final int locatorNumber = i;
+      randomAvailableTCPPortKeepers.get(locatorNumber).release();
+      VM.getVM(i).invoke("start locator", () -> {
+        try {
+          Disconnect.disconnectFromDS();
+          dsProperties = null;
+          Properties props = getDistributedSystemProperties();
+          locator = Locator.startLocatorAndDS(locatorPorts[locatorNumber], new File(""), props);
+          system = locator.getDistributedSystem();
+          cache = ((InternalLocator) locator).getCache();
+          ReconnectDUnitTest.savedSystem = locator.getDistributedSystem();
+          IgnoredException.addIgnoredException(
+              "org.apache.geode.ForcedDisconnectException||Possible loss of quorum");
+        } catch (IOException e) {
+          Assert.fail("unable to start locator", e);
+        }
+      });
+    }
+  }
+
+  @After
+  public void teardown() {
+    for (int i = 0; i < NUM_LOCATORS; i++) {
+      VM.getVM(i).invoke(() -> {
+        InternalLocator locator = InternalLocator.getLocator();
+        if (locator != null) {
+          InternalConfigurationPersistenceService sharedConfig =
+              locator.getConfigurationPersistenceService();
+          if (sharedConfig != null) {
+            sharedConfig.destroySharedConfiguration();
+          }
+          locator.stop();
+        }
+      });
+    }
+    Invoke.invokeInEveryVM(() -> {
+      if (system != null) {
+        system.disconnect();
+      }
+      system = null;
+      cache = null;
+    });
+  }
+
+  public Properties getDistributedSystemProperties() {
+    dsProperties = new Properties();
+    dsProperties.put(MAX_WAIT_TIME_RECONNECT, "20000");
+    dsProperties.put(ENABLE_NETWORK_PARTITION_DETECTION, "true");
+    dsProperties.put(DISABLE_AUTO_RECONNECT, "false");
+    dsProperties.put(ENABLE_CLUSTER_CONFIGURATION, "true");
+    dsProperties.put(USE_CLUSTER_CONFIGURATION, "true");
```

Comment:
You should probably also add `dsProperties.put(HTTP_SERVICE_PORT, "0");` otherwise the test fails in IntelliJ *if* you have `GEODE_HOME` environment variable set. This results in multiple http services attempting to start on the same port (7070) - one in each locator.